modules/rostests/apitests/shell32/ShellExecuteW.cpp

```diff
@@ -40,111 +59,67 @@ START_TEST(ShellExecuteW)
         SysDrive[2] = 0;
     }
     PathAddBackslashW(SysDrive);
+    GetWindowList(&g_winlist);
 
     // TEST #1: Open Control Panel
     hInstance = ShellExecuteW(NULL, L"open", L"rundll32.exe", L"shell32.dll,Control_RunDLL desk.cpl",
                               NULL, SW_SHOWNORMAL);
     ret = (INT)(UINT_PTR)hInstance;
     ok(ret > 31, "TEST #1: ret:%d, LastError: %ld\n", ret, GetLastError());
     trace("TEST #1 ret: %d.\n", ret);
-    if (hInstance)
-    {
-        Sleep(WAIT_SLEEP);
-        // Terminate Window
-        hWnd = FindWindowW(NULL, L"Display Properties");
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
-    }
+    CloseWindow(hInstance, NULL, L"Properties for Display");
 
     // TEST #2: Open Notepad
     hInstance = ShellExecuteW(NULL, L"open", L"notepad.exe", NULL, NULL, SW_SHOWNORMAL);
     ret = (INT)(UINT_PTR)hInstance;
     ok(ret > 31, "TEST #2: ret:%d, LastError: %ld\n", ret, GetLastError());
     trace("TEST #2 ret: %d.\n", ret);
-    if (hInstance)
-    {
-        Sleep(WAIT_SLEEP);
-        // Terminate Window
-        hWnd = FindWindowW(L"Notepad", L"Untitled - Notepad");
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
-    }
+    CloseWindow(hInstance, L"Notepad", L"Untitled - Notepad");
 
     // TEST #3: Open Windows folder
     hInstance = ShellExecuteW(NULL, NULL, WinDir, NULL, NULL, SW_SHOWNORMAL);
     ret = (INT)(UINT_PTR)hInstance;
     ok(ret > 31, "TEST #3: ret:%d, LastError: %ld\n", ret, GetLastError());
     trace("TEST #3 ret: %d.\n", ret);
-    if (hInstance)
-    {
-        Sleep(WAIT_SLEEP);
-        // Terminate Window
-        hWnd = FindWindowW(L"CabinetWClass", WinDir);
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
-    }
+    CloseWindow(hInstance, L"CabinetWClass", PathFindFileNameW(WinDir));
 
     // TEST #4: Open system32 folder
     hInstance = ShellExecuteW(NULL, L"open", SysDir, NULL, NULL, SW_SHOWNORMAL);
     ret = (INT)(UINT_PTR)hInstance;
     ok(ret > 31, "TEST #4: ret:%d, LastError: %ld\n", ret, GetLastError());
     trace("TEST #4 ret: %d.\n", ret);
-    if (hInstance)
-    {
-        Sleep(WAIT_SLEEP);
-        // Terminate Window
-        hWnd = FindWindowW(L"CabinetWClass", SysDir);
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
-    }
+    CloseWindow(hInstance, L"CabinetWClass", PathFindFileNameW(SysDir));
```

Comment:
Are we sure the explorer window is shown between the time `ShellExecuteW` is invoked above and here?